target/linux/ar71xx/image/senao.mk

```diff
@@ -0,0 +1,30 @@
+define Build/senao-factory-image
+	mkdir -p $(KDIR_TMP)/$(1)
+	touch $(KDIR_TMP)/$(1)/FWINFO-OpenWrt-$(REVISION)-$(1)
+	# $(CP) ./$(1)/* $(KDIR_TMP)/$(1)
+	dd if=$(KDIR)/$(1)-kernel.bin \
+		of=$(KDIR_TMP)/$(1)/openwrt-senao-$(1)-uImage-lzma.bin bs=64k conv=sync
+	dd if=$(KDIR)/root.squashfs \
+		of=$(KDIR_TMP)/$(1)/openwrt-senao-$(1)-root.squashfs bs=64k conv=sync
+	( \
+		cd $(KDIR_TMP)/$(1);  \
+		$(TAR) -cz -f $(call factoryname,squashfs,$(1)) * \
+	)
+	rm -rf $(KDIR_TMP)/$(1)
+endef
+
+define Device/ens202ext
```

Comment:
>  Now, regarding sysupgrade I don't know, but the factory image (which my understanding says is the way from vendor->lede, correct me if I'm wrong) has to come out the same as my current recipe has (that is, a separate uImage-lzma.bin and root.squashfs with the exact naming scheme, tar'd and gzipped).

Just to make this more clear. the factory image format is _only_ used to upgrade from OEM firmware to LEDE. LEDE has its own image formats to do an upgrade from your current LEDE version to a later one.

sysupgrade.tar is one of the formats used by LEDE but it is only used if you have a NAND flash chip which might uses UBI. In your case a SPI flash chip is used and you need to use the ["normal" LEDE image format](https://github.com/lede-project/source/blob/master/target/linux/ar71xx/image/Makefile#L110) which is added to the ar71xx default image build recipe. By using ```IMAGES += factory.img``` you keep the sysupgrade.bin and add the factory.img (```+=``` vs. ```:=```).